Add support for Petrified Blood EHP

deathbeam opened the pull request:
This adds support for Petrified Blood life loss prevention to EHP calcs
in similar fashion as Mind over Matter. The prevention gain is checked
against being at low life or not, below low life it will use full value
(e.g 40 base) above full life it will use halved value (Pathfinders for
example utilize this mechanic and recover to full only via flasks).

Signed-off-by: Tomas Slusny <slusnucky@gmail.com>

Fixes #2737

### Link to a build that showcases this PR:

https://pastebin.com/TxUzDmsC (full life Petrified Blood Pathfinder)

### Before screenshot:

![image](https://user-images.githubusercontent.com/5115805/151639363-384af2d7-17c5-4cbd-9ece-251998aa5e58.png)

### After screenshot:

![image](https://user-images.githubusercontent.com/5115805/151639379-8a3e1894-e224-4c7e-8ac5-79ff9a0f8c0c.png)

deathbeam commented on src/Data/Skills/act_str.lua:
```diff
@@ -4929,6 +4929,13 @@ skills["PetrifiedBlood"] = {
+		},
+		["cannot_recover_above_low_life_except_flasks"] = {
+			flag("CannotLeechLife", { type = "GlobalEffect", effectType = "Buff" }, { type="Condition", var="LowLife", neg = true }),
+			flag("NoLifeRegen", { type = "GlobalEffect", effectType = "Buff" }, { type="Condition", var="LowLife", neg = true }),
```
Hmm you are right and i also dont rly want to deal with that, my main goal was the EHP anyway, removed the regen part

Nostrademous commented on src/Modules/CalcSections.lua:
```diff
@@ -1601,6 +1601,28 @@ return {
 			{ modName = { "GuardAbsorbRate", "ChaosGuardAbsorbRate", "GuardAbsorbLimit", "ChaosGuardAbsorbLimit" } },
 		},
 	},
+	{ label = "Prevented life loss", haveOutput = "AnyPreventedLifeLoss",
```
can you just capitalize this to `Prevented Life Loss` to keep format with other rows in that table please?

deathbeam replied:
Sure, updated

deathbeam commented:
Updated the PR and moved Petrified Blood calcuations before guard skill, as based on this post that is the correct order: https://www.pathofexile.com/forum/view-thread/3123794

The actual order btw is (from bottom to top):

Petrified blood, x before life (so mind over matter), energy shield, guard, aegis, frost shield, jinxed juju

So I think also mind over matter is in incorrect position and should be *before* guard skill, but I am not going to fix that here as that is different issue.

Regisle commented:
@deathbeam the whole thing is being reworked anyway in #2390 and its order is correctly applied there, also your list is missing ward

deathbeam commented:
Ah i see. And yea that post is missing info about ward and i was just going based on that.

Wires77 commented:
@deathbeam The calc defences rework PR has been merged into dev, so can you take a look at the conflicts for me and we'll take another look

Regisle commented:
I am in the process of fixing it to work with the new system, as well as giving people an idea of what "degen" they can expect

deathbeam commented:
Closing in favour of #4060 because of the calc defenses rework
